cmake_modules/FindArrow.cmake

```diff
@@ -43,7 +43,7 @@ if ("${ARROW_HOME}" STREQUAL "")
 
     set(ARROW_LIB_NAME ${CMAKE_SHARED_LIBRARY_PREFIX}arrow)
 
-    set(ARROW_SHARED_LIB ${ARROW_LIBDIR}/${ARROW_LIB_NAME}${CMAKE_SHARED_LIBRARY_SUFFIX}${ARROW_SHARED_LIB_SUFFIX})
```

Comment:
Can you revert this? 

`libarrow.so.0` is actually correct.

Reply:
On Mac OS, libarrow.0.0.0.dylib is correct (not libarrow.dylib.0.0.0). I can limit the change to "APPLE" rather than revert it.

https://cmake.org/cmake/help/latest/variable/APPLE.html

ls /usr/local/lib/libarrow*
/usr/local/lib/libarrow.0.0.0.dylib	/usr/local/lib/libarrow.a
/usr/local/lib/libarrow.0.dylib		/usr/local/lib/libarrow.dylib

ls /usr/local/lib/libparquet*
/usr/local/lib/libparquet.1.0.0.dylib	/usr/local/lib/libparquet.a
/usr/local/lib/libparquet.1.dylib	/usr/local/lib/libparquet.dylib

ls /usr/local/lib/libgflags*
/usr/local/lib/libgflags.2.2.1.dylib		/usr/local/lib/libgflags_nothreads.2.2.1.dylib
/usr/local/lib/libgflags.2.2.dylib		/usr/local/lib/libgflags_nothreads.2.2.dylib
/usr/local/lib/libgflags.dylib			/usr/local/lib/libgflags_nothreads.dylib

Reply:
Oh, yes! macOS and Linux have different ordering. Limiting the change to APPLE should be the correct way forward.